src/test/regress/sql/multi_subquery_complex_queries.sql

```diff
@@ -2400,3 +2400,20 @@ FROM
   USING (user_id)
 GROUP BY a.user_id
 ORDER BY 1;
+
+-- queries where column aliases are used
+-- the query is not very complex. join is given an alias with aliases
+-- for each output column
+SELECT k1 
+FROM (
```

Comment:
now non-subquery outer joins also goes through from query pushdown planning (#2481). So, I think it is valuable to add one more test with such a query.

Reply:
Two example queries I've used which might save some time for you. 
The tables will be in the regression test after #2597: 
```SQL
SELECT *
FROM (test_table_1 AS t1
      FULL JOIN test_table_3 AS t2 ON (t1.id = t2.id
                                       AND t1.val1=t2.val1)) k(x, y, z, t)
ORDER BY 1, 2, 3, 4;


SELECT *
FROM (test_table_1 AS t1
      FULL JOIN test_table_3 AS t2 USING (val1,
                                          id)) k(x, y)
ORDER BY 1,
         2;
```

Reply:
added